homotopy-graphics/src/svg.rs

```diff
@@ -1,2 +1,51 @@
+use std::fmt::Write;
+
+use homotopy_core::Generator;
+
+use crate::style::{Color, GeneratorStyle, SignatureStyleData};
+
 pub mod render;
 pub mod shape;
+
+pub fn stylesheet(styles: &impl SignatureStyleData, prefix: &str) -> String {
+    let mut stylesheet = String::new();
+
+    for generator in styles.generators() {
+        let style = styles.generator_style(generator).unwrap();
+
+        writeln!(
+            stylesheet,
+            ".{name} {{ fill: {color}; stroke: {color}; }}",
+            name = class(prefix, generator, "surface"),
+            color = hex(&style.color().lighten(0.1))
+        )
+        .unwrap();
+        writeln!(
+            stylesheet,
+            ".{name} {{ stroke: {color}; }}",
+            name = class(prefix, generator, "wire"),
+            color = hex(&style.color().lighten(0.05))
+        )
+        .unwrap();
+        writeln!(
+            stylesheet,
+            ".{name} {{ fill: {color}; }}",
+            name = class(prefix, generator, "point"),
+            color = hex(&style.color())
+        )
+        .unwrap();
+    }
+
+    stylesheet
+}
+
+pub fn class(prefix: &str, generator: Generator, suffix: &str) -> String {
```

Comment:
Looking at this, it seems we have the same function in manim, tikz, and svg (with the difference that in svg it takes a suffix as well). Maybe we could combine them and move them into `style.rs`.

Reply:
Also I wanted to ask your thoughts about this idea I have had.

Instead of having three CSS classes for every generator
```css
.generator_0_0--point
.generator_0_0--wire
.generator_0_0--surface
```
would it be possible to do this by mixing CSS classes, so something like
```css
.generator_0_0
.generator_0_0 .wire
.generator_0_0 .surface
```

Is it possible to write classes `.wire` and `.surface` which lighten the stroke/fill colour?

Reply:
Good idea, I'll have a look at this.

Reply:
Thanks! That can be a separate PR.

Also related to #550, which is one of the reasons I want to do this so we don't have 9 css classes for every generator.